libglusterfs/src/inode.c

```diff
@@ -1393,12 +1451,17 @@ inode_parent(inode_t *inode, uuid_t pargfid, const char *name)
 
         if (dentry)
             parent = dentry->parent;
-
-        if (parent)
-            __inode_ref(parent, false);
     }
     pthread_mutex_unlock(&table->lock);
 
+    if (parent && !__is_root_with_ref(parent)) {
```

Comment:
Q: does it make sense to unlock the table and right away lock it if this condition is true? Unsure.

Reply:
I actually read about it before implementing it this way, it adds `80-90ns` overhead. IMO, the cases where this condition is false will compensate and overall balance this approach. What do you think?

Reply:
Unsure really. Let's keep it the way you did it, we can always come back to it later. The latency we are concerned about is not how much the 'if' statement takes, nor how much we save by skipping the unlock, lock steps - it's how much contention the table lock has, and if by skipping this we actually complete some work faster - not always clear what's the winning strategy is.

Reply:
Besides the potential performance gains (which I think are very marginal if any at all), this creates a race which is more dangerous.